### ISSUE TITLE
N-Attenuator Combo Class

### ISSUE THREAD

**Issue by ZLLentz** [pcdshub/pcdsdevices#256]
Requested by Koglin for early run 17

The goal is to use multiple attenuators together to increase the granularity by which we can achieve the desired transmission ratio.

Basic algorithm is:
- use N attenuators
- move first attenuator to the closest transmission above the desired transmission
- check the actual transmission, find the new multiplier that the rest of the attenuators need
- move second attenuator to the closest transmission above the desired transmission
- check the actual transmission... etc. etc. repeat
- move last attenuator as close as possible

**Comment by ZLLentz** [comment]
I was thinking making a positioner that has dominion over attenuator objects, so you could pass it into a normal scan and get results.

I wrote this as an N-tenuator, but in practice you do this with 2: your hutch attenuator and the fee att.

**Comment by teddyrendahl** [comment]
Right, but if it is a device then everyone needs to make their special Fee + attenuator class. If it is a function we can just make it `set_combined_attenuation(attenuation, *attenuators)` and it is usable everywhere. Doesn't even necessarily have to be a `bluesky` scan

**Comment by ZLLentz** [comment]
I walked away from github for 5 minutes and had the same thought. This deserves at least a small discussion before anyone dives straight into writing it.

**Comment by ZLLentz** [comment]
Note that making this an external method means you can't directly read the combined attenuation

**Comment by ZLLentz** [comment]
This needs the following before moving to the algorithm:
- FeeAtt class
- a way to force ceiling/floor in main attenuator classes and in the fee attenuator class
